loopy/target/c/codegen/expression.py

```diff
@@ -416,6 +416,10 @@ def map_constant(self, expr, type_context):
                     else "_Complex_I")
             return real + imag*iota
         elif np.isnan(expr):
+            from warnings import warn
+            warn("Consider using `pymbolic.primitives.NaN` instead of `math.nan`."
+                 " The generated code will be equivalent with the added benefit"
+                 " of sound pickling/unpickling of kernel objects.")
```

Comment:
```suggestion
            warn(
                "Encountered 'bare' floating point NaN value. Since NaN != NaN, "
                "this leads to problems with cache retrieval. "
                "Consider using `pymbolic.primitives.NaN` instead of `math.nan`."
                " The generated code will be equivalent with the added benefit"
                " of sound pickling/unpickling of kernel objects.")
```

Reply:
Thanks, fixed!